### PR TITLE
FLAG-69: Improve the performance of the evaluating all flags

### PR DESCRIPTION
**Pull request by ManojLL**
## Description of what I changed

refactor the evaluate all flag opeate in parallel 

Evaluate all flags 10 times
parallel process -> 53, 58 , 73 . 51 , 46, 57 , 58, 63 , 52 => avg 51.1 milliseconds

current module -> 258, 230, 215, 208, 207, 208, 204, 196,202, 198 => avg 212.6 milliseconds

## Issue I worked on

see https://openmrs.atlassian.net/browse/FLAG-69

## Checklist: I completed these to help reviewers :)

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [x] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.
  
  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/patientflags/task/PatientFlagTask.java [hunk 1]
```diff
@@ -126,8 +126,22 @@ public void run() {
 			FlagService flagService = Context.getService(FlagService.class);
 			
 			for (Flag flag : flagService.getAllFlags()) {
-				generatePatientFlags(flag, flagService);
+				Daemon.runInNewDaemonThread(new PatientFlagGenerator(flag));
```
Comment (wikumChamith) [inline api/src/main/java/org/openmrs/module/patientflags/task/PatientFlagTask.java:129]: Can we use streams instead of a for-loop?

Comment (ManojLL) [reply api/src/main/java/org/openmrs/module/patientflags/task/PatientFlagTask.java]: I updated the code

File: api/src/main/java/org/openmrs/module/patientflags/task/PatientFlagTask.java [hunk 2]
```diff
@@ -124,10 +124,22 @@ private static class AllFlagsEvaluator implements Runnable {
...
+		private final Flag flag;
+
+		PatientFlagGenerator(Flag flag){
+			this.flag = flag;
+		}
+
+		@Override
+		public void run() {
+			FlagService service = Context.getService(FlagService.class);
+			generatePatientFlags(flag,service);
```
Comment (wikumChamith) [inline api/src/main/java/org/openmrs/module/patientflags/task/PatientFlagTask.java:142]: Add a space between flag,service:

`generatePatientFlags(flag, service);`

Comment (ManojLL) [reply api/src/main/java/org/openmrs/module/patientflags/task/PatientFlagTask.java]: done

**Pull request merged**